src/SSLSocket.c

```diff
@@ -325,6 +325,21 @@ The user-defined argument optionally defined by SSL_CTX_set_msg_callback_arg() o
 		content_type, (int)len);
 }
 
+void SSL_CTX_keylog_callback(const SSL *ssl, const char *line)
+{
+	const char* sslkeylogfile = getenv("SSLKEYLOGFILE");
+	if (sslkeylogfile)
+	{
+		FILE *fptr;
+
+		fptr = fopen(sslkeylogfile, "at");
+		if (fptr)
+		{
+			fprintf(fptr, "%s\n", line);
```

Comment:
I went through and verified that this is the format that we should be logging, according to the OpenSSL docs:
> The key logging callback is called with two items: the ssl object associated with the connection, and line, a string containing the key material in the format used by NSS for its SSLKEYLOGFILE debugging output. To recreate that file, the key logging callback should log line, followed by a newline. line will always be a NULL-terminated string.